### PR TITLE
ACP2E-1186: Document the 'count' and 'scope' introduction

### PR DESCRIPTION
**Pull request by thiaramus**
## Purpose of this pull request

This pull request (PR) is to update the integration test guide with the newly introduced parameters for data fixtures

## Affected pages

- https://developer.adobe.com/commerce/testing/guide/integration/attributes/data-fixture/
- https://developer.adobe.com/commerce/testing/guide/integration/attributes/data-fixture-before-transaction/

## Links to related PRs or Jira tickets

- https://jira.corp.adobe.com/browse/ACP2E-1186

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My changes follow the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ To set up a date fixture, use the `DataFixture` attribute.
 
 ```php?start_inline=1
 #[
-    DataFixture(string $type, array $data = [], ?string $as = null)
+    DataFixture(string $type, array $data = [], string $count = 1, string $scope = 'default', ?string $as = null)
```
Comment (bubasuma) [inline src/pages/guide/integration/attributes/data-fixture.md:19]: The current order does not match the constructor signature.
```suggestion
    DataFixture(string $type, array $data = [], ?string $as = null, ?string $scope = null, int $count = 1)
```

I am glad that PHP8 introduced named arguments that allows passing values to optional arguments independently of their positions

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 2]
```diff
@@ -77,6 +81,76 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
...
+{
+   #[
+      DataFixture(ProductFixture::class, count: 3),
+   ]
+   public function testGetProductsCount(): void
+   {
+      $fixtures = DataFixtureStorageManager::getStorage();
+      $product1 = $fixtures->get('product1');
+      $product2 = $fixtures->get('product2');
+      $product3 = $fixtures->get('product3');
```
Comment (bubasuma) [inline src/pages/guide/integration/attributes/data-fixture.md]: no alias will be assigned to the fixtures if the alias is not explicitly defined

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 3]
```diff
@@ -77,6 +81,76 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
...
+   }
```
Comment (bubasuma) [inline src/pages/guide/integration/attributes/data-fixture.md:101]:
```suggestion
   #[
        DataFixture(CategoryFixture::class, ['is_anchor' => 0], 'category1'),
        DataFixture(CategoryFixture::class, ['parent_id' => '$category1.id$'], 'category2'),
        DataFixture(ProductFixture::class, ['category_ids' => ['$category1.id$']])
        DataFixture(ProductFixture::class, ['category_ids' => ['$category2.id$']], count: 2)
    ]
    public function testAddCategoryFilter(): void
    {
        $fixtures = DataFixtureStorageManager::getStorage();
        $category1 = $fixtures->get('category1');
        $category2 = $fixtures->get('category2');
        $collection = $this->collectionFactory->addCategoryFilter($category2);
        $this->assertEquals(2, $collection->getSize());
        $collection = $this->collectionFactory->addCategoryFilter($category1);
        $this->assertEquals(1, $collection->getSize());
    }
```

**Review by bubasuma — CHANGES_REQUESTED** [review]

**Review by bubasuma — APPROVED** [review]

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 4]
```diff
@@ -26,6 +26,10 @@ To set up a date fixture, use the `DataFixture` attribute.
    -  Name of a class that implements `Magento\TestFramework\Fixture\DataFixtureInterface` or `Magento\TestFramework\Fixture\RevertibleDataFixtureInterface`.
 -  **data**
    -  The optional array of data passed on to the fixture.
+-  **count**
+   -  The optional number of entities of the same kind and configuration should this fixture generate.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:30]:
```suggestion
   - Available to developers using the [2.4-develop](https://github.com/magento/magento2/tree/2.4-develop) branch only. Scheduled for release in 2.4.6.
   -  The optional number of entities of the same kind and configuration should this fixture generate.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 5]
```diff
@@ -26,6 +26,10 @@ To set up a date fixture, use the `DataFixture` attribute.
...
+-  **scope**
+   -  The optional store view, website or store group identifier, from which the fixture will generate the entity.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:32]:
```suggestion
   - Available to developers using the [2.4-develop](https://github.com/magento/magento2/tree/2.4-develop) branch only. Scheduled for release in 2.4.6.
   -  The optional store view, website or store group identifier, from which the fixture will generate the entity.
```

**Review by jeff-matthews — COMMENTED** [review]
@thiaramus, I haven't done a full editorial review, but I made a couple of suggestions per our Slack conversation to identify the `count` and `scope` parameters as dev-only. This isn't the only way we can do it, but it seemed like the most direct and simple approach. What do you think?

File: src/pages/guide/integration/attributes/data-fixture-before-transaction.md [hunk 6]
```diff
@@ -21,6 +21,10 @@ The data fixture before transaction attribute is an extension of [Data Fixture a
    -  The fully qualified name of a class that implements `Magento\TestFramework\Fixture\DataFixtureInterface` or `Magento\TestFramework\Fixture\RevertibleDataFixtureInterface`.
 -  **data**
    -  The optional array of data passed on to the fixture.
+-  **count**
+   -  The optional number of entities of the same kind and configuration should this fixture generate.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture-before-transaction.md:25]:
```suggestion
   -  The optional number of entities of the same kind and configuration that this fixture should generate.
```

File: src/pages/guide/integration/attributes/data-fixture-before-transaction.md [hunk 7]
```diff
@@ -21,6 +21,10 @@ The data fixture before transaction attribute is an extension of [Data Fixture a
...
+-  **scope**
+   -  The optional store scope name, from which the fixture will generate the entity.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture-before-transaction.md:27]:
```suggestion
   -  The optional store scope name from which the fixture will generate the entity.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 8]
```diff
@@ -77,6 +83,82 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
 }
 ```
 
+### Specifying the number of instances of data fixture to generate
+
+<InlineAlert variant="info" slots="text" />
+
+At this time, the parameter is only available for Adobe Commerce developer, and will become publicly available with Adobe Commerce 2.4.6 release.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:90]:
```suggestion
At this time, the parameter is available to developers using the [2.4-develop](https://github.com/magento/magento2/tree/2.4-develop) branch only. Scheduled for release in 2.4.6.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 9]
```diff
@@ -26,6 +26,12 @@ To set up a date fixture, use the `DataFixture` attribute.
    -  Name of a class that implements `Magento\TestFramework\Fixture\DataFixtureInterface` or `Magento\TestFramework\Fixture\RevertibleDataFixtureInterface`.
 -  **data**
    -  The optional array of data passed on to the fixture.
+-  **count**
+   - Available to developers using the [2.4-develop](https://github.com/magento/magento2/tree/2.4-develop) branch only. Scheduled for release in 2.4.6.
+   -  The optional number of entities of the same kind and configuration should this fixture generate.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:31]:
```suggestion
   -  The optional number of entities of the same kind and configuration that this fixture should generate.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 10]
```diff
@@ -26,6 +26,12 @@ To set up a date fixture, use the `DataFixture` attribute.
...
+-  **scope**
+   - Available to developers using the [2.4-develop](https://github.com/magento/magento2/tree/2.4-develop) branch only. Scheduled for release in 2.4.6.
+   -  The optional store view, website or store group identifier, from which the fixture will generate the entity.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:34]:
```suggestion
   -  The optional store view, website, or store group identifier from which the fixture will generate the entity.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 11]
```diff
@@ -77,6 +83,82 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
 }
 ```
 
+### Specifying the number of instances of data fixture to generate
+
+<InlineAlert variant="info" slots="text" />
+
+At this time, the parameter is only available for Adobe Commerce developer, and will become publicly available with Adobe Commerce 2.4.6 release.
+
+Sometimes we need to generate several instances of a data fixture with exactly the same configuration.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:92]:
```suggestion
Sometimes, we need to generate several instances of a data fixture with exactly the same configuration.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 12]
```diff
@@ -77,6 +83,82 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
...
+For such cases, we can use the `count` parameter, and set its value to the desired number of instances.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:93]:
```suggestion
For such cases, we can use the `count` parameter and set its value to the desired number of instances.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 13]
```diff
@@ -77,6 +83,82 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
...
+   {
+      $fixtures = DataFixtureStorageManager::getStorage();
+      $product1 = $fixtures->get('product1');
+      $product2 = $fixtures->get('product2');
+      $product3 = $fixtures->get('product3');
+   }
+}
+```
+
+The generated fixtures will be assigned aliases product1, product2 and product3 respectively.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:135]:
```suggestion
The generated fixtures will be assigned aliases product1, product2, and product3 (respectively).
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 14]
```diff
@@ -77,6 +83,82 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
...
+
+### Specifying the store scope for the data fixture
+
+<InlineAlert variant="info" slots="text" />
+
+At this time, the parameter is only available for Adobe Commerce developer, and will become publicly available with Adobe Commerce 2.4.6 release.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:141]:
```suggestion
At this time, the parameter is available to developers using the [2.4-develop](https://github.com/magento/magento2/tree/2.4-develop) branch only. Scheduled for release in 2.4.6.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 15]
```diff
@@ -77,6 +83,82 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
...
+
+If you need to instruct the system to execute a data fixture in the scope of a specific store view, you can set the `scope` parameter value to the valid store view, website or store group identifier.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:143]:
```suggestion
If you need to instruct the system to execute a data fixture in the scope of a specific store view, you can set the `scope` parameter value to the valid store view, website, or store group identifier.
```

File: src/pages/guide/integration/attributes/data-fixture.md [hunk 16]
```diff
@@ -77,6 +83,82 @@ class QuoteTest extends \PHPUnit\Framework\TestCase
...
+
+In the example below, we create a new store with the `store2` identifier and a product. Then we create a guest cart under the `store2` scope and add a created product to it.
```
Comment (jeff-matthews) [inline src/pages/guide/integration/attributes/data-fixture.md:145]:
```suggestion
In the following example, we create a new store with the `store2` identifier and a product. Then, we create a guest cart under the `store2` scope and add a created product to it.
```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]

**Review by jeff-matthews — APPROVED** [review]
@thiaramus, just let me know when you're ready to merge.

**Comment by thiaramus** [comment]
> @thiaramus, just let me know when you're ready to merge.

It is now ready. Thank you, Jeff!

**Review by jeff-matthews — APPROVED** [review]

**Pull request merged**